provider/resource.go

```diff
@@ -3575,6 +3525,62 @@ func Provider() tfbridge.ProviderInfo {
 	return prov
 }
 
+func defaultAzureLocation(p tfshim.Provider, rgRegionMap map[string]string) func(res *tfbridge.PulumiResource) (interface{}, error) {
+	return func(res *tfbridge.PulumiResource) (interface{}, error) {
+		// In here we will fetch the resource group property from this resource and
+		// use it to query the Azure API and return the resource group's location. We
+		// maintain a little cache to avoid querying the API too many times. Note that
+		// it's possible (likely) during previews that the location will be unknown, so
+		// we special logic to propagate likewise unknown location values.
+		if rg, has := res.Properties["resourceGroupName"]; has {
+			if rg.IsComputed() || rg.IsOutput() {
+				return tfbridge.TerraformUnknownVariableValue, nil
+			}
+			if rg.IsString() {
+				rgName := rg.StringValue()
+				rgRegion, has := rgRegionMap[rgName]
+				if !has {
+					rgRes := p.ResourcesMap().Get("azurerm_resource_group")
+					contract.Assert(rgRes != nil)
+					importer := rgRes.Importer()
+					contract.Assert(importer != nil)
+					states, err := importer("azurerm_resource_group",
+						fmt.Sprintf("/subscriptions/_/resourceGroups/%s",
+							rg.StringValue()), p.Meta())
+					if err != nil {
+						return nil, err
+					}
+					switch stateLen := len(states); stateLen {
+					case 0:
+						rgRegion = tfbridge.TerraformUnknownVariableValue
+					case 1:
+						state, err := p.Refresh("azurerm_resource_group", states[0], nil)
```

Comment:
I have added the `, nil)` here in response to https://github.com/pulumi/pulumi-terraform-bridge/pull/1312. Otherwise this is just a refactor into a function to remove indentation and clarify captured state.